hardhat.config.ts

```diff
@@ -36,12 +37,20 @@ if (PK) {
   };
 }
 
-if (["rinkeby", "mainnet"].includes(argv.network) && INFURA_KEY === undefined) {
+if (
+  ["rinkeby", "mainnet"].includes(argv.network) &&
```

Comment:
What about xdai?

Reply:
If network is `xdai`, then there is a default node set. Otherwise, there must be either an Infura key or a node url.